### PR TITLE
Add distribution page and data inside Ecosystem

### PR DESCRIPTION
**Pull request by theletterf**
Following https://github.com/open-telemetry/opentelemetry.io/discussions/3812, we agreed on trying a single page approach for distributions, with an emphasis on stating that distributions are optional.

Some pending matters:

- There are around 30 distributions inside the vendors table. Should we move them?
- Should we add a `type` field to the `distributions.yaml` file for the types described [here](https://opentelemetry.io/docs/concepts/distributions/#what-is-a-distribution)?

Preview: https://deploy-preview-3854--opentelemetry.netlify.app/ecosystem/distributions/

image

### CODE REVIEW

**Comment by theletterf** [comment]
Preview: https://deploy-preview-3854--opentelemetry.netlify.app/ecosystem/distributions/

**Review by cijothomas — APPROVED** [review]

**Comment by svrnm** [comment]
> There are around 30 distributions inside the vendors table. Should we move them?

Let's keep them for now, we can remove them later, maybe let those who have provided a contact know that they can update.

> Should we add a type field to the distributions.yaml file for the types described [here](https://opentelemetry.io/docs/concepts/distributions/#what-is-a-distribution)?

I lean towards saying _no_, because who is going to validate which one they provide.

The OSS flag is kind of misleading, because the distribution is OSS, but the backend might not and the things built into the distribution are often not working without that particular backend. But I don't have a better solution at hand.

File: content/en/ecosystem/distributions.md
Comment (svrnm) [inline content/en/ecosystem/distributions.md:15]: Add a sentence here that calls out specifically, that a distribution is not a "must have", and that many organisations and vendors just live fine without one (especially for language SDKs, collector might be the exception here)

File: content/en/ecosystem/distributions.md
Comment (svrnm) [inline content/en/ecosystem/distributions.md:26]:
```suggestion
- Link to the main page of your distribution
- Link to the documentation that explains how to use the distribution
- List the components your distribution contains
- GitHub handle or email address as a point of contact so that we can reach out in case we have questions
```

**Review by svrnm — CHANGES_REQUESTED** [review]
a few more comments

File: content/en/ecosystem/distributions.md
Comment (theletterf) [reply content/en/ecosystem/distributions.md:19]: How about this? https://github.com/open-telemetry/opentelemetry.io/pull/3854/commits/9dff456af19688f51b2fbff9f45323a94624a239

File: data/ecosystem/distributions.yaml
Comment (mhausenblas) [inline data/ecosystem/distributions.yaml:7]: AWS Distro for OpenTelemetry (ADOT)

File: data/ecosystem/distributions.yaml
Comment (mhausenblas) [inline data/ecosystem/distributions.yaml:8]: Technically, the ADOT product homepage is https://aws.amazon.com/otel/

**Review by mhausenblas — CHANGES_REQUESTED** [review]
LGTM, thanks!

File: data/ecosystem/distributions.yaml
Comment (theletterf) [reply data/ecosystem/distributions.yaml:7]: @mhausenblas Feel free to edit my branch directly!

File: data/ecosystem/distributions.yaml
Comment (theletterf) [reply data/ecosystem/distributions.yaml:8]: Ditto!

File: content/en/ecosystem/distributions.md
Comment (svrnm) [inline content/en/ecosystem/distributions.md:4]: fix that description

File: data/ecosystem/distributions.yaml
Comment (svrnm) [inline data/ecosystem/distributions.yaml:11]: I wonder if we would even accept non-OSS distributions, what would be the purpose of that?

File: content/en/ecosystem/distributions.md
Comment (svrnm) [inline content/en/ecosystem/distributions.md:18]:
```suggestion
The list below contains a sample of OpenTelemetry distributions and the
component they customize. Note that those are third-party projects and are in no way validated or endorsed by the OpenTelemetry community.
```

**Review by cartermp — CHANGES_REQUESTED** [review]
Will need to add the following too (and remove them as whatever they're classified as):

https://github.com/open-telemetry/opentelemetry.io/blob/main/data/registry/instrumentation-go-honeycomb.yml
https://github.com/open-telemetry/opentelemetry.io/blob/main/data/registry/instrumentation-java-honeycomb.yml
https://github.com/open-telemetry/opentelemetry.io/blob/main/data/registry/instrumentation-dotnet-honeycomb.yml
https://github.com/open-telemetry/opentelemetry.io/blob/main/data/registry/instrumentation-js-node-honeycomb.yml

File: data/ecosystem/distributions.yaml
Comment (theletterf) [reply data/ecosystem/distributions.yaml:11]: Hmmm, good point. Let me remove that field.

File: content/en/ecosystem/distributions.md
Comment (theletterf) [reply content/en/ecosystem/distributions.md:18]: Edited your suggestion to change "below" to "following" as per https://developers.google.com/style/word-list#below

File: content/en/ecosystem/distributions.md
Comment (chalin) [inline content/en/ecosystem/distributions.md:5]:
```suggestion
```

File: content/en/ecosystem/distributions.md
Comment (chalin) [inline content/en/ecosystem/distributions.md:4]: How about emphasizing OSS here?

```suggestion
description: >-
  List of open source OpenTelemetry distributions maintained by third parties.
```

**Review by chalin — COMMENTED** [review]
Some initial remarks, see inline.

File: content/en/ecosystem/distributions.md
Comment (theletterf) [reply content/en/ecosystem/distributions.md]: Added!

File: content/en/ecosystem/distributions.md
Comment (chalin) [inline content/en/ecosystem/distributions.md:17]: I'd suggestion putting this note inside a `{{% alert title="Note" color="warning" %}}` possibly with revisted prose like this:

> OpenTelemetry **_does not_ validate or endorse** the third-party distributions listed below. The list is provided as a convenience for the community.

File: content/en/ecosystem/distributions.md
Comment (chalin) [inline content/en/ecosystem/distributions.md:24]:
```suggestion
To have your distribution listed, submit a PR with an entry added to
the
```

File: content/en/ecosystem/distributions.md
Comment (chalin) [inline content/en/ecosystem/distributions.md:46]:
```suggestion
{{% alert title="Notes" color="info" %}}

- If you provide external integration of OpenTelemetry for any kind of library,
service, or app, then consider
[adding it to the registry](/ecosystem/registry/adding).

- If you adopt OpenTelemetry for observability as an end user and do not provide
any kind of services around OpenTelemetry, see [Adopters](/ecosystem/adopters).

- If you provide a solution that consumes OpenTelemetry to offer observability to
end users, see [Vendors](/ecosystem/vendors).
```

File: content/en/ecosystem/distributions.md
Comment (chalin) [reply content/en/ecosystem/distributions.md:5]: Hmm, this was marked as resolved, but the alias is still there.

File: content/en/ecosystem/distributions.md
Comment (chalin) [inline content/en/ecosystem/distributions.md:43]:
```suggestion
```

File: content/en/ecosystem/distributions.md
Comment (chalin) [inline content/en/ecosystem/distributions.md:47]:
```suggestion
```

File: content/en/ecosystem/distributions.md
Comment (theletterf) [reply content/en/ecosystem/distributions.md]: Hmm, let me check that

Comment (theletterf) [reply content/en/ecosystem/distributions.md]: Ah, now: https://github.com/open-telemetry/opentelemetry.io/pull/3854/commits/27604c5b125bad84c4b55f13aa29a929f6e63405

**Comment by theletterf** [comment]
@svrnm @chalin Looking nicer now, with all the distros I know of. Should we sort by component first perhaps, and then name? https://deploy-preview-3854--opentelemetry.netlify.app/ecosystem/distributions/

**Comment by chalin** [comment]
> @svrnm @chalin Looking nicer now, with all the distros I know of. Should we sort by component first perhaps, and then name? https://deploy-preview-3854--opentelemetry.netlify.app/ecosystem/distributions/

Yes, that sounds like a good idea to me.

**Comment by chalin** [comment]
@theletterf - I can't get it to work either. _Maybe_ it could have to do with the fact that `components` is a list. Sorting is meant to be stable in Hugo, so we should be able to get the result that we want. Let's just sort by name for now 🤷🏼‍♂️.

**Comment by chalin** [comment]
Sorry, ... but I'm having second thoughts about part of the design here. _Maybe_ we should indeed "enhance" the Registry to also include distributions? We could keep this Distributions page (with the warnings etc), but instead of the table, it could link into the registry with the appropriate query parameters. WDYT @theletterf @svrnm et al.?

**Comment by theletterf** [comment]
@chalin If we do that, should we also do it for the integrations? Another option could be to re-do the shortcodes so that they parse the registry for a subset of data (and add a field stating whether something is a distribution or not). That would accomplish keeping a centralized place for external stuff instead of increasing fragmentation — which I guess is your primary concern?

**Comment by chalin** [comment]
Originally, the registry was meant to list OTel "components" AFAIK (@austinlparker can confirm), so earlier in this PR-review history I thought that it was a good idea to pull out the distributions.

I think that the first discussion we (@open-telemetry/docs-approvers) need to have is whether the registry could/should be generalized to include integrations as well as reintroduce distributions.

With the recent enhancements that @svrnm made to the registry, it's a very nice I/F to use to look up "anything OTel related" :). If we agree to have integrations and distributions be part of the registry (and I'm not yet fully convinced that we should) we could "simply" update **Type**s to include "Distribution" and "Integration" (and replace "Any component" by just "Any")

I wouldn't redo the shortcodes to filter anything, I'd just link to the registry using appropriate query parameters. My 2 cents.

Thoughts @svrnm et al.?

**Comment by chalin** [comment]
> The way we're presenting the registry right now, it's a repository of components and utilities. Distributions _wrap_ and _bundle_ those components, so having them in the Registry feels weird to me. ...

To me too.

> To add to the complexity, we have the Adopters and Vendors pages inside Ecosystem, which clearly cannot go into the Registry. They seem to follow a separate need, that is, to communicate who's using and endorsing OTel.

Yes, makes sense.

> Possible solutions I see: ...
> Which one is your favorite?

Let's go with what this PR provides :). Thanks for entertaining and discussing alternatives. Later, if the table of distributions becomes to wieldy, we can also consider using a registry-like I/F to support filtering etc.

**Review by chalin — APPROVED** [review]
⭐

**Comment by svrnm** [comment]
Commenting on the registry, data-tables, independent pages discussions, here is something I have in my head:

Eventually I would like to see us moving ALL the things otel related (integrations, vendors, distributions) into the registry. We can still keep those pages (and restrict the registry to the relevant elements), but this requires a little bit more steps inbetween, so as of right now, I want to merge this PR, have the page with the distributions and fix the rest later.

**Review by svrnm — APPROVED** [review]

**Comment by svrnm** [comment]
@mhausenblas @cartermp can you double check that your change requests have been applied and provide approval? Thanks

**Review by mhausenblas — APPROVED** [review]
Thanks, LGTM!

**Review by cartermp — APPROVED** [review]

**Pull request merged**
